.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,203 @@
+name: run-tests
+on:
+  workflow_call:
+    inputs:
+      node-matrix:
+        description: "An array of node versions against which the tests should be run"
+        required: false
+        type: string
+        default: "['lts/*']"
+
+jobs:
+  # Step 1: We have a number of dependencies that we can build now and cache
+  #         for future use, to save time later.
+  build-dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      # First thing's first; we need to get the code.
+      - uses: actions/checkout@v2
+      # Next, we need to make sure we're using the right versions of Node and Java.
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 'lts/*' # Always use Node LTS for building dependencies.
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '8' # Always use Java v1.8 for building dependencies.
+      # Install/build dependencies.
+      - run: yarn
+      - run: yarn build
+      # Create a tarball.
+      - run: npm pack
+      # Upload the `dist` directory as an artifact so the unit test jobs have it.
+      - uses: actions/upload-artifact@v3
+        with:
+          name: unit-test-dist
+          path: ./dist
+      # Upload the tarball as an artifact so the smoke test jobs have it.
+      - uses: actions/upload-artifact@v3
+        with:
+          name: smoke-test-tarball
+          path: ./salesforce-sfdx-scanner-*.tgz
+  # Step 2: Run our tests.
+  # Step 2A: Run the unit tests.
+  unit-tests:
+    strategy:
+      # By default, if any job in a matrix fails, all other jobs are immediately canceled. This makes the jobs run
+      # to completion instead.
+      fail-fast: false
+      matrix:
+        node: ${{ fromJson(inputs.node-matrix) }}
+        os: [ubuntu-latest, windows-2019]
+    runs-on: ${{ matrix.os }}
+    needs: build-dependencies
+    steps:
+      # Check out the code.
+      - uses: actions/checkout@v2
+      # Make sure we're using the right versions of Node and Java.
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }} # Node version is a matrix.
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '11' # For now, Java version is hardcoded. It could be changed to a matrix in the future.
```

Comment:
Shouldn't this be `8`?

Reply:
I'm pretty sure this should be 11. But we could matrixify it so it uses 8 and 11.